datalad/distribution/publish.py

```diff
@@ -98,12 +98,20 @@ def has_diff(ds, refspec, remote, paths):
               remote_branch_name, remote)
     current_commit = ds.repo.get_hexsha()
     within_ds_paths = [p for p in paths if p['path'] != ds.path]
-    if within_ds_paths:
+    commit_differ = current_commit != ds.repo.get_hexsha(remote_ref)
+    # yoh: not sure what "logic" was intended here for comparing only
+    # some files.  By now we get a list of files, if any were changed,
+    # from the commit on remote, and somehow diff says below that they didn't differ...
+    # but if commit is different -- there must be differences and we
+    # should publish. otherwise now skips publishing root dataset
+    # although its master is behind by 1 commit.  Moreover there could
+    # be an empty commit -- shouldn't we publish then???
+    if not commit_differ and within_ds_paths:
```

Comment:
this is my main concern about my fix... not sure if this doesn't obliterate some magic, which should've worked while the fix had to be implemented elsewhere (e.g. in filtering the paths away, so here we would end up with no paths, thus pushing on irrespective according to the older logic)

Reply:
So what do you think @mih?